File: src/main/java/nz/co/aetheric/sameness/generators/XMLGregorianCalendarGenerator.java
package nz.co.aetheric.sameness.generators;

import com.pholser.junit.quickcheck.generator.GenerationStatus;
import com.pholser.junit.quickcheck.generator.Generator;
import com.pholser.junit.quickcheck.generator.InRange;
import com.pholser.junit.quickcheck.random.SourceOfRandomness;

import javax.xml.datatype.DatatypeConfigurationException;
import javax.xml.datatype.DatatypeFactory;
import javax.xml.datatype.XMLGregorianCalendar;

/**
 * TODO: What is the purpose of this class?
 * <p>Author: <a href="http://gplus.to/tzrlk">Peter Cummuskey</a></p>
 */
@SuppressWarnings("WeakerAccess")
public class XMLGregorianCalendarGenerator extends Generator<XMLGregorianCalendar> {

	private final GregorianCalendarGenerator calendars = new GregorianCalendarGenerator();
	private final DatatypeFactory datatypeFactory;

	public XMLGregorianCalendarGenerator() throws DatatypeConfigurationException {
		super(XMLGregorianCalendar.class);
		datatypeFactory = DatatypeFactory.newInstance();
	}

	public void configure(InRange range) {
		calendars.configure(range);
	}

	@Override
	public XMLGregorianCalendar generate(SourceOfRandomness random, GenerationStatus status) {
		return datatypeFactory.newXMLGregorianCalendar(calendars.generate(random, status));
	}

}

File: src/main/kotlin/nz/co/aetheric/sameness/generators/XMLGregorianCalendarGenerator.kt
package nz.co.aetheric.sameness.generators

import com.pholser.junit.quickcheck.generator.GenerationStatus
import com.pholser.junit.quickcheck.generator.Generator
import com.pholser.junit.quickcheck.generator.InRange
import com.pholser.junit.quickcheck.random.SourceOfRandomness

import javax.xml.datatype.DatatypeConfigurationException
import javax.xml.datatype.DatatypeFactory
import javax.xml.datatype.XMLGregorianCalendar

/**
 * TODO: What is the purpose of this class?
 *
 * Author: [Peter Cummuskey](http://gplus.to/tzrlk)
 */
class XMLGregorianCalendarGenerator() : Generator<XMLGregorianCalendar>(XMLGregorianCalendar::class.java) {

	private val calendars = GregorianCalendarGenerator()
	private val datatypeFactory: DatatypeFactory = DatatypeFactory.newInstance()

	fun configure(range: InRange)
			= calendars.configure(range)

	override fun generate(random: SourceOfRandomness, status: GenerationStatus)
			= datatypeFactory.newXMLGregorianCalendar(calendars.generate(random, status))

}
